backend/apps/ecommerce/tests.py

```diff
@@ -0,0 +1,220 @@
+        self.assert_permission_error(response)
+
+        # Authorized user should be able to retrieve order
+        response = self.query(query, user=self.indok_user)
+        self.assertResponseNoErrors(response)
+
+    def test_resolve_user_orders(self):
+        query = """
+                query userOrders {
+                    userOrders {
+                        orderId
+                            product {
+                                id
+                                name
+                            }
+                            user {
+                                username
+                            }
+                            quantity
+                            totalPrice
+                            paymentStatus
+                            date
+                    }
+                }
+                """
+
+        # Unauthorized user should not be able to retrieve orders
+        response = self.query(query)
+        self.assert_permission_error(response)
+
+        # Authorized user should be able to retrieve orders
+        response = self.query(query, user=self.indok_user)
+        self.assertResponseNoErrors(response)
+
+        # Fetching content of response
+        content = json.loads(response.content)
+
+        # There is one order in the database
+        self.assertEqual(len(content["data"]["userOrders"]), 1)
+
+
+class VippsApiMock:
+    def initiate_payment(order):
+        return "https://www.youtube.com/watch?v=dQw4w9WgXcQ"  # Just a random URL
+
+    def capture_payment(order, method):
+        print("Order: ", order, "\nMethod: ", method)
+
+
+class EcommerceMutationsTestCase(EcommerceBaseTestCase):
+    """
+    Testing all mutations for ecommerce
+    """
+
+    def test_create_product(self):
+        """
+        THIS TEST MUST BE REWRITTEN!
+        """
+
+        query = """
+          mutation {
+            createProduct {
+                product {
+                    id
+                    name
+                    description
+                    price
+                    totalQuantity
+                    maxBuyableQuantity
+                }
+                ok
+                }
+            }
+        """
+        response = self.query(query)
+        self.assertResponseNoErrors(response)
+
+        data = json.loads(response.content)["data"]
+        response_product = data["createProduct"]["product"]
+        product = Product.objects.get(pk=response_product["id"])
+        if product is None:
+            self.assertIsNotNone(product, msg="Expected product after creation, got None")
+        else:
+            self.assertTrue(data["createProduct"]["ok"])
+            product.price = int(product.price)
+            response_product["price"] = int(response_product["price"])
+            self.deep_assert_equal(response_product, product)
+
+    @unittest.mock.patch.object(InitiateOrder, "vipps_api", VippsApiMock)
+    def test_initiate_order(self):
```

Comment:
This should be split into several, smaller tests for easier debugging and parallelization of tests